### PR TITLE
Reset counters and reset_sw_pll_pfd when external clock goes invalid in clockgen

### PR DESCRIPTION
**Pull request by shuchitak**
also, addresses https://github.com/xmos/lib_xua/issues/500

### CODE REVIEW

**Comment by xross** [comment]
During discussion we concluded this might not be the correct way forward. See #500. More investigation to follow.

**Comment by shuchitak** [comment]
> During discussion we concluded this might not be the correct way forward. See #500. More investigation to follow.

Hi @xross, I've changed this as discussed. Also tested and ensured that this fixes the very negative ferror issue.

**Comment by shuchitak** [comment]
> Running this through the sw_usb_audio regression shows some new issues (http://srv-bri-jcim0:8080/job/XMOS/job/sw_usb_audio/job/test_dig_rx_mclk_fix/4/). Will continue to work on this.

There was an issue in initialising timeNextClockDetection for the sw_pll case which was causing the adat_input tests to fail on MacOS. After fixing them, sw_usb_audio smoke run looks good (http://srv-bri-jcim0:8080/job/XMOS/job/sw_usb_audio/job/test_dig_rx_mclk_fix/7/pipeline-console/?selected-node=565).

I propose we merge this and https://github.com/xmos/lib_xua/pull/504 and run it through the sw_usb_audio nightly.

File: lib_xua/src/core/clocking/clockgen.xc [hunk 1]
```diff
@@ -524,11 +540,25 @@ void clockGen ( streaming chanend ?c_spdif_rx,
 #if (XUA_SPDIF_RX_EN)
                     /* Returns 1 if valid clock found */
                     valid = validSamples(spdifCounters, CLOCK_SPDIF);
+#if XUA_USE_SW_PLL
+                    if(clockValid[CLOCK_SPDIF] && (valid == 0))
+                    {
+                        reset_sw_pll_pfd = 1;
```
[the remark this answers is not in the document]
Comment (shuchitak) [reply lib_xua/src/core/clocking/clockgen.xc:550]: done.
